internal/compile/compile.go

```diff
@@ -1170,6 +1206,47 @@ func (fcomp *fcomp) assignSequence(pos syntax.Position, lhs []syntax.Expr) {
 	}
 }
 
+// addr emits code for a chain of x.f and a[i] operations such as a.f[i].g[j].
+//
+// In Addressing mode, all nonrecursive calls to addr() (a chain of ATTR
+// and INDEX ops) must be followed by one of ADDRESS, SETFIELD,
+// SETINDEX, or VALUE:
```

Comment:
This really confused me until I read the interpreter's code for `VALUE` a second time: it's a no-op for values that don't implement `Variable`. I think that should be mentioned here, too.